src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -501,7 +501,80 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
+      //new Predef.doubleToDouble(x).intValue() -> x.toInt()
+      //new Predef.doubleToDouble(x).longValue() -> x.toLong()
+      //new Predef.doubleToDouble(x).floatValue() -> x.toFloat()
+      //new Predef.doubleToDouble(x).doubleValue() -> x.toDouble()
+      //
+      //methods on Float
+      //new Predef.float2Float(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.float2Float(x).isInfinite() -> java.lang.Double.isInfinity(x)
+      //methods on Number
+      //new Predef.float2Float(x).byteValue() -> x.toByte()
+      //new Predef.float2Float(x).shortValue() -> x.toShort()
+      //new Predef.float2Float(x).intValue() -> x.toInt()
+      //new Predef.float2Float(x).longValue() -> x.toLong()
+      //new Predef.float2Float(x).floatValue() -> x.toFloat()
+      //new Predef.float2Float(x).doubleValue() -> x.toDouble()
+      case Apply(Select(Apply(boxing, params), methodName), Nil)
+        if (boxing.symbol == currentRun.runDefinitions.Predef_double2Double ||
+          boxing.symbol == currentRun.runDefinitions.Predef_float2Float) &&
+          params.size == 1 &&
+          allFloatingMethodsToRewrite.contains(methodName) =>
+        val newTree =
+          if (doubleAndFloatRedirectMethods.contains(methodName)) {
+            val cls =
+              if (boxing.symbol == currentRun.runDefinitions.Predef_double2Double)
+                definitions.BoxedDoubleClass
+              else definitions.BoxedFloatClass
+
+            val targetMethod = cls.companionModule.info.decl(doubleAndFloatRedirectMethods(methodName))
+            gen.mkMethodCall(targetMethod, params)
+          } else {
+            gen.mkMethodCall(params.head.symbol, javaNumberConversions(methodName), Nil)
+          }
+        super.transform(localTyper.typedPos(tree.pos)(newTree))
 
+      //(x:Int).hashCode is transformed to scala.Int.box(x).hashCode()
+      //(x:Int).toString is transformed to scala.Int.box(x).toString()
+      //
+      //rewrite
+      // scala.Int.box(x).hashCode() ->  java.lang.Integer.hashCode(x)
+      // scala.Int.box(x).toString() ->  java.lang.Integer.toString(x)
+      // similarly for all primitive types
+      case Apply(Select(Apply(box @ Select(boxer, _), params), methodName), Nil)
+        if objectMethods.contains(methodName) &&
+          params.size == 1
+          && currentRun.runDefinitions.isBox(box.symbol)
+      =>
+        val cls = boxer.symbol.companion
+        val target = {
+          if (cls == definitions.BooleanClass) BoxedBooleanClass
+          else if (cls == definitions.CharClass) BoxedCharacterClass
+          else if (cls == definitions.ByteClass) BoxedByteClass
+          else if (cls == definitions.ShortClass) BoxedShortClass
+          else if (cls == definitions.IntClass) BoxedIntClass
+          else if (cls == definitions.LongClass) BoxedLongClass
+          else if (cls == definitions.FloatClass) BoxedFloatClass
+          else if (cls == definitions.DoubleClass) BoxedDoubleClass
```

Comment:
Wow, surprisingly it seems this mapping doesn't exist anywhere in the compiler... 🤯

Reply:
Isn't it https://github.com/scala/scala/blob/aeee8f0b2f9f970d07dd33fdb49e93fa1140e103/src/reflect/scala/reflect/internal/Definitions.scala#L99 ?